src/System.Net.HttpListener/src/System/Net/Windows/WebSockets/HttpWebSocket.Windows.cs

```diff
@@ -14,24 +14,17 @@ namespace System.Net.WebSockets
 {
     internal static partial class HttpWebSocket
     {
-        private static readonly ArraySegment<byte> s_EmptyPayload = new ArraySegment<byte>(new byte[] { }, 0, 0);
         private static readonly Random s_keyGenerator = new Random();
-        private static readonly bool s_httpSysSupportsWebSockets = (Environment.OSVersion.Version >= new Version(6, 2));
-
-        internal static ArraySegment<byte> EmptyPayload
-        {
-            get { return s_EmptyPayload; }
-        }
 
         internal static Task<HttpListenerWebSocketContext> AcceptWebSocketAsync(HttpListenerContext context,
             string subProtocol,
             int receiveBufferSize,
             TimeSpan keepAliveInterval,
             ArraySegment<byte> internalBuffer)
         {
-            ValidateOptions(subProtocol, receiveBufferSize, WebSocketBuffer.MinSendBufferSize, keepAliveInterval);
-            ValidateArraySegment(internalBuffer, nameof(internalBuffer));
-            WebSocketBuffer.Validate(internalBuffer.Count, receiveBufferSize, WebSocketBuffer.MinSendBufferSize, true);
+            ValidateOptions(subProtocol, receiveBufferSize, MinSendBufferSize, keepAliveInterval);
```

Comment:
> I'm not sure I understand this change: why are we changing from the .Net Framework port to unify with a new, managed implementation.

We are unifying the managed and windows implementation to fix managed bugs. In all these situations, as you will see, we prefer the Windows code. So in essence, what I've done is move non-Windows specific code to the PAL layer, removed the managed implementation code that does the same thing - thus Windows and managed have the same behaviour by definition as they have the same code.

> Is there an assumption here that the buffer bounds are the same between using the HTTP.SYS APIs and the managed one? (In this case it may - which would make the change OK - just confirming.)

Previously some of the inputs in the unit tests I've added would hit Debug assertions as they are invalid in the managed stream.

Basically we want almost complete Windows compatability in the managed implementation. I believe one of the reasons for this is that we plan to use the managed implementation of HttpListener in UWP on Windows.